pilot/pkg/xds/ep_filters.go

```diff
@@ -134,6 +136,16 @@ func (b *EndpointBuilder) EndpointsByNetworkFilter(endpoints []*LocLbEndpointsAn
 	return filtered
 }
 
+func (b *EndpointBuilder) gatewaysByNetwork() map[model.NetworkID][]*model.Gateway {
```

Comment:
There's a lot of converting back and forth between slice and map. Can we expose the underlying map in push.Gateways, maybe comment a warning not to modify it.

Reply:
My plan was to eliminate that in the follow-on PR that will modify this load balancing algorithm.